interceptors/logging/interceptors.go

```diff
@@ -47,40 +50,102 @@ func (c *reporter) PostCall(err error, duration time.Duration) {
 	if err != nil {
 		logger = logger.With("error", fmt.Sprintf("%v", err))
 	}
-	logger.With(c.opts.durationFieldFunc(duration)...).Log(c.opts.levelFunc(code), fmt.Sprintf("finished %s %s call", c.kind, c.typ))
+
+	if !c.opts.shouldLogRequest(interceptors.FullMethod(c.service, c.method), err) {
+		logger.With(c.opts.durationFieldFunc(duration)...).Log(c.opts.levelFunc(code), fmt.Sprintf("finished %s %s call", c.kind, c.typ))
+	} else {
```

Comment:
instead of else we can do `return` See https://thanos.io/contributing/coding-style-guide.md/#control-structure-prefer-early-returns-and-avoid-else